src/hwcaps.c

```diff
@@ -101,12 +91,19 @@ static uint32_t GetElfHwcapFromGetauxval(uint32_t hwcap_type) {
   dlclose(libc_handle);
   return ret;
 }
-#endif  // defined(HWCAPS_ANDROID_MIPS_OR_ARM)
+#elif defined(CPU_FEATURES_OS_LINUX_OR_ANDROID)
+// On Regular Linux we simply use getauxval.
+#include <dlfcn.h>
+#include <sys/auxv.h>
```

Comment:
May you should check via cmake that "sys/auxv.h" exists.
see https://cmake.org/cmake/help/latest/module/CheckIncludeFile.html